### PR TITLE
chore: simpleSupply and simpleFail, two tranches with base distributor and default assessor

### PR DESCRIPTION
**Pull request by charleenfei**
- also optimisations for creating TestUsers

### CODE REVIEW

File: src/core/test/system/base_system.sol [hunk 1]
```diff
@@ -53,21 +53,31 @@ contract BaseSystemTest is TestSetup, Math, DSTest {
...
             admin_ = address(admin);
             rootAdmin.relyBorrowAdmin(admin_);
 
             juniorInvestor = new Investor(address(juniorOperator), currency_, address(juniorERC20));
             juniorInvestor_ = address(juniorInvestor);
+
             WhitelistOperator juniorOperator = WhitelistOperator(address(juniorOperator));
             juniorOperator.relyInvestor(juniorInvestor_);
+
+            if (senior_) {
```
Comment (manuelpolzhofer) [inline src/core/test/system/base_system.sol]: 👍

File: src/core/test/system/lender/base/supply_two_tranche.sol [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        assertEq(currency.balanceOf(address(junior)), 0);
+        assertEq(currency.balanceOf(address(senior)), 10 ether);
+        assertEq(senior.debt(), 10 ether);
+
+        hevm.warp(now + 1 days);
+
+        assertEq(senior.debt(), 10.5 ether);
+
+        // change the ratio, senior can still supply
+        assessor.file("minJuniorRatio" , 0);
```
Comment (manuelpolzhofer) [inline src/core/test/system/lender/base/supply_two_tranche.sol]: I think we should add this method to the test deployer later.

File: src/core/test/system/base_system.sol [hunk 3]
```diff
@@ -53,21 +53,31 @@ contract BaseSystemTest is TestSetup, Math, DSTest {
         rootAdmin.relyLenderAdmin(address(this), senior_);
     }
 
-    function createTestUsers() public {
-            borrower = new Borrower(address(shelf), address(lenderDeployer.distributor()), currency_, address(pile));
+    function createTestUsers(bool senior_) public {
```
Comment (manuelpolzhofer) [inline src/core/test/system/base_system.sol]: It would have been easier by adding just adding.
```javascript
function createTestUsers() public {
  bool deploySenior = false;
  createTestUsers(deploySenior);
}
```

both ways are fine I think

File: src/core/test/system/lender/base/supply_two_tranche.sol [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        seniorInvestor.doSupply(sSupplyAmount);
+    }
+
+    function testFailSimpleSupply() public {
+        uint investorBalance = 100 ether;
+        currency.mint(juniorInvestor_, investorBalance);
+        currency.mint(seniorInvestor_, investorBalance);
+
+        uint jSupplyAmount = 80 ether;
+        uint sSupplyAmount = 20 ether;
```
Comment (manuelpolzhofer) [inline src/core/test/system/lender/base/supply_two_tranche.sol]: maybe let's do it the other way round.

junior normally is around 10%-20%

File: src/core/test/system/base_system.sol (hunk 3 above)
Comment (ilinzweilin) [reply src/core/test/system/base_system.sol]: yes, agree with Manuel. 
Also helos to avid merge conflicts

File: src/core/test/system/system.t.sol [hunk 5]
```diff
@@ -23,8 +23,8 @@ contract STest is BaseSystemTest {
     Hevm public hevm;
     
     function setUp() public {
-        baseSetup("whitelist", "default", false);
-        createTestUsers();
+        baseSetup("whitelist", "switchable", false);
+        createTestUsers(false);
```
Comment (ilinzweilin) [inline src/core/test/system/system.t.sol]: why not using dafault dustributor for the system tests?

File: src/core/test/system/borrower/withdraw.t.sol [hunk 6]
```diff
@@ -25,8 +25,8 @@ contract WithdrawTest is BaseSystemTest {
         bytes32 juniorOperator_ = "whitelist";
         bytes32 distributor_ = "default";
         baseSetup(juniorOperator_, distributor_, false);
-        createTestUsers();
-        distributor = DefaultDistributor(address(lenderDeployer.distributor()));
+        createTestUsers(false);
+        distributor = SwitchableDistributor(address(lenderDeployer.distributor()));
```
Comment (ilinzweilin) [inline src/core/test/system/borrower/withdraw.t.sol]: why did you change to Switchable here? Should be the default

File: src/core/test/system/borrower/lock.sol [hunk 7]
```diff
@@ -23,7 +23,7 @@ contract LockTest is BaseSystemTest {
         bytes32 juniorOperator_ = "whitelist";
         bytes32 distributor_ = "default";
         baseSetup(juniorOperator_, distributor_, false);
-        createTestUsers();
+        createTestUsers(false);
```
Comment (ilinzweilin) [inline src/core/test/system/borrower/lock.sol]: would be really cool if you just overload the function 🙈

File: src/core/test/system/lender/base/supply_two_tranche.sol [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+        assertEq(seniorERC20.balanceOf(seniorInvestor_), sSupplyAmount);
+
+        uint minJuniorRatio = 8 * 10**26;
+        assessor.file("minJuniorRatio" , minJuniorRatio);
+
+        // doesn't break ratio, junior supply now 90 and senior supply 20
+        uint jAdditionalSupply = 10 ether;
+        juniorInvestor.doSupply(jAdditionalSupply);
+
+        assertEq(currency.balanceOf(address(junior)), jSupplyAmount + jAdditionalSupply);
```
Comment (ilinzweilin) [inline src/core/test/system/lender/base/supply_two_tranche.sol]: please use safeMath / Tinlake Math here
